exchanges/okex/okex_test.go

```diff
@@ -1,300 +1,1587 @@
 package okex
 
 import (
+	"fmt"
 	"testing"
+	"time"
 
 	"github.com/thrasher-/gocryptotrader/common"
 	"github.com/thrasher-/gocryptotrader/config"
 	"github.com/thrasher-/gocryptotrader/currency/pair"
 	"github.com/thrasher-/gocryptotrader/currency/symbol"
 	exchange "github.com/thrasher-/gocryptotrader/exchanges"
+	"github.com/thrasher-/gocryptotrader/exchanges/okgroup"
 )
 
-var o OKEX
-
 // Please supply you own test keys here for due diligence testing.
 const (
 	apiKey                  = ""
 	apiSecret               = ""
+	passphrase              = ""
+	OKGroupExchange         = "OKEX"
 	canManipulateRealOrders = false
 )
 
-func TestSetDefaults(t *testing.T) {
-	o.SetDefaults()
-	if o.GetName() != "OKEX" {
-		t.Error("Test Failed - Bittrex - SetDefaults() error")
+var testSetupRan bool
+var o = OKEX{}
+var spotCurrency = pair.NewCurrencyPairWithDelimiter(symbol.BTC, symbol.USDT, "-").Pair().Lower().String()
+
+func TestSetDefaults(t *testing.T) {
+	if o.Name != OKGroupExchange {
+		o.SetDefaults()
+	}
+	if o.GetName() != OKGroupExchange {
+		t.Errorf("Test Failed - %v - SetDefaults() error", OKGroupExchange)
+	}
+	TestSetup(t)
+	t.Parallel()
+}
+
+func TestSetRealOrderDefaults(t *testing.T) {
+	TestSetDefaults(t)
+	if areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("Ensure canManipulateRealOrders is true and your API keys are set")
+	}
+}
+
+func TestSetup(t *testing.T) {
+	if testSetupRan {
+		return
+	}
+	if o.APIKey == apiKey && o.APISecret == apiSecret &&
+		o.ClientID == passphrase && apiKey != "" && apiSecret != "" && passphrase != "" {
+		return
+	}
+	o.ExchangeName = OKGroupExchange
+	cfg := config.GetConfig()
+	cfg.LoadConfig("../../testdata/configtest.json")
+
+	okexConfig, err := cfg.GetExchangeConfig(OKGroupExchange)
+	if err != nil {
+		t.Fatalf("Test Failed - %v Setup() init error", OKGroupExchange)
+	}
+
+	okexConfig.AuthenticatedAPISupport = true
+	okexConfig.APIKey = apiKey
+	okexConfig.APISecret = apiSecret
+	okexConfig.ClientID = passphrase
+	okexConfig.WebsocketURL = o.WebsocketURL
+	o.Setup(okexConfig)
+	testSetupRan = true
+}
+
+func areTestAPIKeysSet() bool {
+	if o.APIKey != "" && o.APIKey != "Key" &&
+		o.APISecret != "" && o.APISecret != "Secret" {
+		return true
+	}
+	return false
+}
+
+func testStandardErrorHandling(t *testing.T, err error) {
```

Comment:
I wonder if it would be better to have this return the error to allow for further checking 
That way we don't end up with examples like https://github.com/thrasher-/gocryptotrader/pull/252/files#diff-1405017cb73c41ce881d3e670589ee58R124 where areTestAPIKeysSet() is being double checked

Reply:
Yeah definitely do not want to double check things! I'll go over the cases. I think any test that requires more than whats in `testStandardErrorHandling` shouldn't use it and have its own error handling rather than returning the error. Thanks for picking that up